subsys/net/lib/capture/Kconfig

```diff
@@ -0,0 +1,95 @@
+# Capture network packets
+
+# Copyright (c) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+config NET_CAPTURE
+	bool "Network packet capture support"
+	select NET_L2_VIRTUAL
+	select NET_L2_VIRTUAL_MGMT
+	select NET_L2_IPIP
+	select NET_MGMT
+	select NET_MGMT_EVENT
+	select NET_CONTEXT_NET_PKT_POOL
+	help
+	  This option allows user to capture network packets.
+	  The captured packets are sent to another host for processing.
+	  User can use network packet analyzer like Wireshark to
+	  process the packets.
+	  The captured network packets are sent using IPIP tunnel
+	  as a payload in UDP datagrams.
+
+if NET_CAPTURE
+
+config NET_CAPTURE_PKT_COUNT
+	int "How many network packets to allocate for capture"
+	default 4
+	help
+	  This tells how many net_pkt to allocate for capturing
+	  network traffic. Each network frame sent or received
+	  will allocate one net_pkt for network metadata.
+	  Each net_pkt will contain a list of net_buf's that contain
+	  the actual network data.
+
+config NET_CAPTURE_BUF_COUNT
+	int "How many network buffers to allocate for capture"
+	default 16
+	help
+	  This tells how many net_buf to allocate for capturing
+	  network traffic.
+
+config NET_CAPTURE_DEVICE_COUNT
+	int "Number of network capture devices"
+	default 1
+	help
+	  Number of network capture devices. Usually one is enough but
+	  if one needs to send captured data to multiple different devices,
+	  then you need to increase the value.
+
+config NET_CAPTURE_TUNNEL_REMOTE_ADDR
+	string "Outer: Remote IP address of the tunnel interface"
+	help
+	  The value tells the tunnel remote outer endpoint IP address.
+	  The local address is resolved automatically at runtime.
+	  The IP address can be either IPv4 or IPv6 address.
+	  Example:
+	  192.0.2.2
+	  2001:db8::2
+
+config NET_CAPTURE_TUNNEL_MY_ADDR_PORT
```

Comment:
Yes, this was done on purpose. The use is / should be explained on the documentation commit. I created this option so that application can use it and do not need to create the Kconfig entry for this purpose.

Reply:
But then sample application configures those options and there is no visible change to the user. It might be confusing when user will configure those and still don't get automatically configured tunnel. If they are defined here, then lets at least use those options in sample application. Otherwise I don't see the reason to keep them if they are just "ghost" options and never used.

Reply:
The documentation has now this
```
It is not necessary to use ``net-shell`` for configuring the monitoring.
The network capture API functions can be called by the application if needed.
For that purpose, the capture Kconfig file contains these configure options

.. csv-table::
   :widths: auto

   :option:`CONFIG_NET_CAPTURE_TUNNEL_REMOTE_ADDR`
   :option:`CONFIG_NET_CAPTURE_TUNNEL_MY_ADDR_PORT`
   :option:`CONFIG_NET_CAPTURE_TUNNEL_PEER_ADDR_PORT`

which can be set by the application ``prj.conf`` file and then given as
parameters when calling :c:func:`net_capture_setup` function.
These Kconfig options are currently not used by the capture sample
application or by the net_capture API but are provided as a convenience for
the end user.
```
but perhaps it is confusing to the reader. The motivation for doing it like this was that we could have consistent naming for the config options even if they are not directly used by the service itself.

Reply:
I will remove the extra options as they can be confusing to the end user.